### ISSUE TITLE
The Alias in openTSDB does not work

### ISSUE THREAD

**Issue by biomack** [grafana/grafana#67176]
Update, May 31, 2023
Thank you everyone for reporting this. Here is our confirmation.

**What is happening:**
OpenTSDB alias does not support tags that contain characters such as `-` or `.`.

**What can be done about this?**
We have identified this as a limitation of the OpenTSDB alias. OpenTSDB support as a plugin is under "maintenance only." We will be moving it out of core Grafana in the future to be supported by the community. In the meantime, fixing the bug that alias does not support `-` or `.` character does not fall into the category of maintenance. 

## What happened:
When we try use alias in Grafana higher then 8.3.7 it just not work

## What you expected to happen:
-
## How to reproduce it (as minimally and precisely as possible):
in 9.4.7
<img width="1243" alt="Screenshot 2023-04-25 at 11 59 31" src="https://user-images.githubusercontent.com/45202893/234163806-b5666d18-842e-43df-b6da-266684262a67.png">

## Environment:
- Grafana version: 9.3.6-9.4.7
- Data source type & version: 2.3
- OS Grafana is installed on: k8s
- User OS & Browser:chrome, firefox
- Grafana plugins:OpenTSDB (core)
- Others:

**Comment by mellieA** [comment]
@biomack was this working for you prior to 8.3.7? 

I can't tell if [this](https://github.com/grafana/grafana/pull/12336) or something similar was ever implemented.

**Comment by biomack** [comment]
this was added in grafana 5 I think  but stopped working after update. 
Yes it is working in 8.3.7

**Comment by anchalchaudhary** [comment]
Is this issue picked for a fix yet? Running into the same issue. This was working until v8.4.11.

**Comment by virenderdubey** [comment]
+1, Facing a similar issue on v8.5. Please see if this can be picked.

**Comment by Govardhan10** [comment]
Facing a similar issue from v8.5.x and above. To give some more context tag values like are ```appId``` , ```conatiner_id``` but some other values like ```event-type```. Like it is able to populates simple words and some special character but not some other special characters.

**Comment by bohandley** [comment]
@biomack [[<alias>]] is deprecated as variable syntax.

[Please see this link for more information](https://grafana.com/docs/grafana/latest/dashboards/variables/variable-syntax/#variable-syntax).

The following variable syntax is supported.

- `$varname` This syntax is easy to read, but it does not allow you to use a variable in the middle of a word. Example: apps.frontend.$server.requests.count

- `${var_name}` Use this syntax when you want to interpolate a variable in the middle of an expression.
- `${var_name:<format>}` This format gives you more control over how Grafana interpolates values. Refer to [Advanced variable format options](https://grafana.com/docs/grafana/latest/dashboards/variables/variable-syntax/#advanced-variable-format-options) for more detail on all the formatting types.

**Comment by Govardhan10** [comment]
The following is not working with the above documentation
<img width="791" alt="Screenshot 2023-05-31 at 9 48 51 AM" src="https://github.com/grafana/grafana/assets/37766383/93c542d0-4c16-47de-8dd5-b7a0665b0342">

**Comment by JozoVilcek** [comment]
@bohandley It is not that OpenTSDB does not support character `-` but more accurately it did stop supporting it. This is a regression. Can you confirm that this broken feature will not fall under the maintenance?

Are you able give some support in identifying which commit did create the regression. It might be easier for someone from community to help with the fix.

**Comment by bohandley** [comment]
Good point, @JozoVilcek, and I hear you. We stopped supporting it. It will not fall under maintenance. 

OpenTSDB has used the template variable replace function for many years now, as far back as 6.1.6 I believe. https://github.com/grafana/grafana/commit/e31a3a64e19cb8896207c1633c75ef5b22d44872

The issue is in the replace function not supporting the characters, so we would either have to argue to change this or write a new replace function. 

The replace function is here
https://github.com/grafana/grafana/blob/59dc1956378a9668613ff5d0e59c5b8de68db665/public/app/features/templating/template_srv.ts#L221-L226

The regex is here
https://github.com/grafana/grafana/blob/59dc1956378a9668613ff5d0e59c5b8de68db665/public/app/features/variables/utils.ts#L24

**Comment by bohandley** [comment]
@grafana/dashboards-squad Do you have any context on when we stopped supporting variables that include `-` and `.`?

**Comment by adamyeats** [comment]
Closing due to a long inactivity. Please re-open if the issue still persists in newer versions of Grafana!
